src/views/LotteryV2/components/BuyTicketsModal.tsx

```diff
@@ -0,0 +1,299 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import BigNumber from 'bignumber.js'
+import { ethers } from 'ethers'
+import { Modal, Text, Flex, HelpIcon, BalanceInput, Ticket, useTooltip, Skeleton } from '@pancakeswap/uikit'
+import { useTranslation } from 'contexts/Localization'
+import { useWeb3React } from '@web3-react/core'
+import { getFullDisplayBalance } from 'utils/formatBalance'
+import { getCakeAddress } from 'utils/addressHelpers'
+import { BIG_ZERO } from 'utils/bigNumber'
+import { useAppDispatch } from 'state'
+import { usePriceCakeBusd, useLottery } from 'state/hooks'
+import { fetchUserTicketsAndLotteries } from 'state/lottery'
+import useTheme from 'hooks/useTheme'
+import useTokenBalance, { FetchStatus } from 'hooks/useTokenBalance'
+import useApproveConfirmTransaction from 'hooks/useApproveConfirmTransaction'
+import { useCake, useLotteryV2Contract } from 'hooks/useContract'
+import useToast from 'hooks/useToast'
+import UnlockButton from 'components/UnlockButton'
+import ApproveConfirmButtons, { ButtonArrangement } from 'views/Profile/components/ApproveConfirmButtons'
```

Comment:
Not that I suggest to fix it right away, but this should probably be moved out of Profile if we reuse it in other places

Reply:
100%. I'll add this to follow-up tasks, I think it got moved into Profile in some of the recent component refactoring work.